ports/nrf/mphalport.c

```diff
@@ -50,6 +54,14 @@
 #include "soc/nrfx_coredep.h"
 #endif
 
+#ifndef MICROPY_HW_STDIN_BUFFER_LEN
+#define MICROPY_HW_STDIN_BUFFER_LEN 1024
```

Comment:
This seems large, especially since nrf parts don't usually have that much RAM.

Can we drop it down to 512 like rp2?

I know it's currently 1024 in master, but the new code in this PR uses `ringbuf_free(&stdin_ringbuf)` to never overflow this buffer, and also the CFG_TUD_CDC_RX_BUFSIZE/CFG_TUD_CDC_TX_BUFSIZE values are now 256 (in this PR) vs 64 (in master), so there's already more buffering.